### DISCUSSION TITLE
Upgrade App to using "Modern Auth"?

### DISCUSSION THREAD

**Discussion by Mattray50** [moralmunky/Home-Assistant-Mail-And-Packages#726] [General]
Is there any chance that you are going to update this to be able to use "Modern Authentication" in the near future? I have all my email accounts in MS Office365 and they have shut down Basic Auth and for a very good reason (Hackers). I believe GMail will also soon disable Basic Auth as well.

I think this is a great app and would love to see it to stay safe and secure in the future.

**Comment by firstof9** [comment]
Do you have any links to documentation on this?

**Comment by Mattray50** [comment]
Hi Chris,

Here is a link for Microsofts announcement. https://techcommunity.microsoft.com/t5/exchange-team-blog/basic-authentication-deprecation-in-exchange-online-september/ba-p/3609437https://techcommunity.microsoft.com/t5/exchange-team-blog/basic-authentication-deprecation-in-exchange-online-september/ba-p/3609437

And one from Google.  https://support.google.com/accounts/answer/6010255https://support.google.com/accounts/answer/6010255

I believe Google will still allow “App Passwords” but not sure for how long.

Matt Raymond
On Sep 28, 2022, 9:51 AM -0700, Chris ***@***.***>, wrote:

Do you have any links to documentation on this?

**Comment by firstof9** [comment]
The changes outlined by Microsoft have nothing to do with IMAP, and google has no plans to retire App Passwords.

**Comment by luberk8** [comment]
My Gmail is showing this:
Less secure app access
Some apps and devices use less secure sign-in technology, which makes your account vulnerable. You can turn off access for these apps, which we recommend, or turn it on if you want to use them despite the risks. Google will automatically turn this setting OFF if it’s not being used.
This setting is no longer available.

Any ideas how to get it to work?

**Comment by firstof9** [comment]
If you can't create app passwords at this time, you'll need to wait for oauth support.

**Comment by eliwide** [comment]
You can get basic auth re-enabled until the end of the year on MS Exchange:

https://techcommunity.microsoft.com/t5/exchange-team-blog/basic-authentication-deprecation-in-exchange-online-september/ba-p/3609437

**Comment by Xellinus** [comment]
So do we have an OAuth version in the works? Google won't let you have "less secure apps" enabled AND two factor support, which is required for App Passwords. So it's a no go here

**Reply by firstof9** [reply]
Still a WIP

**Comment by firstof9** [comment]
Some progress
![image](https://user-images.githubusercontent.com/1105672/211214378-b38ca7da-800d-4512-a370-e155ba854e4d.png)
![image](https://user-images.githubusercontent.com/1105672/211214406-46deaa17-c0cb-4059-9efe-4e1971df9036.png)

**Reply by baf** [reply]
@firstof9 Thanks for your work on this. Any more progress since your comment? I'd be happy to test out a dev branch.

**Comment by firstof9** [comment]
Not at this time.

**Reply by baf** [reply]
Okay; thanks. I'll keep a subscription to this topic in case any are forthcoming.

**Comment by BMWfan** [comment]
Hey together, is there any progress? I would like to use a mailbox of office365.

**Reply by firstof9** [reply]
It's proving more difficult than anticipated, sadly.

**Reply by BMWfan** [reply]
mh, what do you think when we could count with it?

**Reply by firstof9** [reply]
As my free time as of late has dwindled, I couldn't say when.

**Reply by BMWfan** [reply]
clear, i understand that. If i can help by doing some test or so, pls let me know.
